### ISSUE TITLE
Nolonger compiles JIT.h not found

### ISSUE THREAD

**Issue by Eforen** [lsegal/my_toy_compiler#28]
bison -d -o parser.cpp parser.y
parser.y: warning: 48 shift/reduce conflicts [-Wconflicts-sr]
g++ -c `llvm-config --cppflags` -std=c++11 -o parser.o parser.cpp
g++ -c `llvm-config --cppflags` -std=c++11 -o codegen.o codegen.cpp
In file included from codegen.cpp:2:0:
codegen.h:16:38: fatal error: llvm/ExecutionEngine/JIT.h: No such file or directory
 #include <llvm/ExecutionEngine/JIT.h>
                                      ^
compilation terminated.
Makefile:27: recipe for target 'codegen.o' failed
make: *** [codegen.o] Error 1

**Comment by rdebath** [comment]
You need to add an llvm version number to this eg:

```sh
$ llvm-config-3.5 --version
3.5.0
$ make LLVMCONFIG=llvm-config-3.5
...
g++ -o parser ...
$
```

BTW: Works for me.

**Comment by joaopaulocl** [comment]
JIT.h has changed in version 3.5 and above. You gonna need to replace it to MCJIT.h in codegen.h.
After that, it is also needed to cast unique_ptr in line 36 of codegen.cpp, just like this:
ExecutionEngine *ee = EngineBuilder((unique_ptr<Module>)module).create();
In my case, to properly compile everything, I had to install libz-dev and libncurses5-dev libncursesw5-dev (use apt-get).

**Comment by lsegal** [comment]
Patches to update this are accepted!

**Comment by willmtemple** [comment]
Should be resolved now that #29 is merged.

**Comment by lsegal** [comment]
Resolving.
